### PR TITLE
add first iteration of the unit testing guidelines

### PR DESCRIPTION
**Pull request by birkirpalmason**

### CODE REVIEW

File: Front-end/Testing.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+2. Try to break function down into separate chunks if the test is beginning to feel bloated.
+
+
+#### Example 
+
+```typescript
+// helper.js 
+function areWeOld(ages: number[]) {
+  if(!age ||!Array.isArray(age)) return [];
+  const oldPeopleAge = 30;
```
Comment (einaralex) [inline Front-end/Testing.md]: 😂

Comment (danielpall) [reply Front-end/Testing.md]: :older_man:

File: Front-end/Testing.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+An exception to this is an _index.js_ file, where the filename should be identical to the folder in which the _index.js_ file lives.
+
+```
+Forms/
+  Login.js
+  Login.test.js
+  Forms.test.js
+  index.js
+```
```
Comment (karljohann) [inline Front-end/Testing.md]: For the app we're using the structure `Forms/__tests__/Forms.test.js`. The snapshots are saved in a folder under `__tests__/__snapshots__` and other files that we use (specific mocks etc.) under `__tests__` as well.

Comment (karljohann) [reply Front-end/Testing.md]: In that case, all js files under `__tests__` are parsed for tests, so the file might as well be named `Forms/__tests__/index.js`. That's maybe not what we want but an option.

File: Front-end/Testing.md (hunk 1 above)
Comment (karljohann) [reply Front-end/Testing.md]: 😭

File: Front-end/Testing.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+  Login.js
+  Login.test.js
+pages/ 
+  about.js
+pages.test/
+  about.test.js
+```
+
+## What should I test? 
+
```
Comment (einaralex) [inline Front-end/Testing.md]: Do we want to leave a link to [this reference sheet](https://github.com/sapegin/jest-cheat-sheet#matchers)? 
It gives you a quick suggestion on how to test different types.

**Pull request merged**